src/devtools/tests/devtools.test.tsx

```diff
@@ -356,7 +356,9 @@ describe('ReactQueryDevtools', () => {
     expect(queries[2]?.textContent).toEqual(query3Hash)
 
     // Wait for the queries to be resolved
-    await sleep(70)
```

Comment:
hm, why don't we just wait for:

```
await waitFor(() => rendered.getByText('query-1-result query-2-result query-3-result'))
```

?

because thats what we print on the screen, and thats what we should see before we proceed. if we see those texts, we know all 3 queries have been loaded ...

this also means we can likely make the sleeps in the queryFns much slower and thus the test itself faster

Reply:
I tried doing it the way you mentioned, but the occasional flaky test problem still persists. On more investigating I found out that the awaiting wasn't the problem, the problem is the `dataUpdatedAt` in query Cache is not incredibly accurate down to milliseconds. 


The test that's failing is the one that sorts the queries according to their `dataUpdatedAt`, in the devtools. So when I kept the gap between every query 10ms (awaiting the queries for 10, 30, and 20ms respectively), all the queries had the same `dataUpdatedAt` time

![image](https://user-images.githubusercontent.com/21277179/143901080-dd44b51b-68d2-4cfe-8058-595f04681750.png)

It only started to show different timestamps when the gap between the queries was ~40ms and the `dataUpdatedAt` in the cache was as expected (query2 resolves at last after query3 and query1).

Reply:
ah interesting. this could actually be a side-effect of how we now do batching and because we're using `useSyncExternalStorage` now. I've also seen that in some other tests, I had to increase the wait time because things were batched otherwise.